src/substrait-to-d3.js

```diff
@@ -0,0 +1,296 @@
+'use strict';
+
+const { SubstraitParser } = require('./substrait-parser');
+const d3 = require('d3');
+
+// Populating map with nodes information
+function populateNodes(plan, nodes) {
+    nodes.set(plan.id, plan);
+    for (let i = 0; i < plan.inputs.length; ++i) {
+        populateNodes(plan.inputs[i], nodes);
+    }
+}
+
+// Building graph from Substrait Plan
+function buildGraph(plan) {
+    let nodes = new Map();
+    populateNodes(plan.inputs[0], nodes);
+    let edges = [];
+    nodes.forEach((value) => {
+        for (let i = 0; i < value.inputs.length; ++i) {
+            edges.push({ 'source': value.inputs[i].id, 'target': value.id })
+        }
+    })
+    return {
+        'nodes': nodes,
+        'edges': edges,
+    };
+}
+
+// Processing nodes for d3JS forcedSimulation's format
+function processNodes(nodes) {
+    let processedNodes = []
+    nodes.forEach((value) => {
+        processedNodes.push({ name: value.id, type: value.type });
+    });
+    return processedNodes;
+}
+
+// Processing edges for d3JS forcedSimulation's format
+function processEdges(nodes, links) {
+    let processedLinks = [];
+    for (let i = 0; i < links.length; ++i) {
+        let source = nodes.findIndex(j => j.name === links[i].source);
+        let target = nodes.findIndex(j => j.name === links[i].target);
+        processedLinks.push({ source, target });
+    }
+    return processedLinks;
+}
+
+// Specifying color of graph nodes
+function nodeColor(nodeType) {
+    switch (nodeType) {
+        case 'root':
+            return "#4D2205";
+        case 'project':
+            return "#0E4D05";
+        case 'read':
+            return "#052748";
+        case 'join':
+            return '#46054D'
+    }
+}
+
+// Specifying Font-Awesome icons for graph nodes
+function nodeIcon(nodeType) {
+    switch (nodeType) {
+        case 'root':
+            return '\uf0ce';
+        case 'project':
+            return '\uf12b';
+        case 'read':
+            return '\uf1c0';
+        case 'join':
+            return '\uf126';
+    }
+}
+
+function typeToLabel(nodeType) {
+    if (nodeType == 'root') {
+        return 'Sink';
+    } else {
+        return nodeType[0].toUpperCase() + nodeType.substring(1);
+    }
+}
+
+// Drawing Graph using d3JS
+function drawGraph(pre_nodes, pre_links) {
+    let width = 960,
+        height = 375;
```

Comment:
Eventually we won't want this to be hard-coded.  Though that is quite a tricky problem to solve :)  Can save for another day.

What we can hard-code is an svg viewbox (assuming we always expect a consistent aspect ratio which is maybe ok for now).  Then the width and height could be assigned in the `html` (probably via a class) and eventually be made dynamic (based on available screen space).

Reply:
Currently, using a auto-resizable svg viewbox by maintaining the original aspect ratio, does this approach looks good?